### PR TITLE
Add LRUCache for modules

### PR DESCRIPTION
**Pull request by robgodfrey-ft**
Resolution of module metadata involves  extracted metadata and readme information from zip archives. This fix adds a Least Recently Used Cache to cache modules to improve performance.

### CODE REVIEW

**Comment by avatarworf** [comment]
Do we need to set the TTL or other params for this lru cache to be initialised

File: lib/puppet_forge_server/backends/directory.rb [hunk 1]
```diff
@@ -82,25 +84,43 @@ def parse_dependencies(metadata)
...
       options = ({:with_checksum => true}).merge(options)
       modules = []
       Dir["#{@module_dir}/**/#{file_name}"].each do |path|
+        modules << get_module(path, options)
+      end
+      modules
+    end
+
+    def get_module(path, options)
+      forge_module = nil
```
Comment (avatarworf) [inline lib/puppet_forge_server/backends/directory.rb:98]: Should this be an empty list instead of nil

Comment (robgodfrey-ft) [reply lib/puppet_forge_server/backends/directory.rb:98]: The function either returns a single module or nil. I don't think an empty list would be appropriate here.

**Comment by robgodfrey-ft** [comment]
> Do we need to set the TTL or other params for this lru cache to be initialised

It works without a TTL parameter, and not sure whether setting one would achieve anything useful in this situation.

**Pull request merged**